# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/html/nsGenericHTMLFrameElement.cpp
# Commit: 656b57e594de
# Full Hash: 656b57e594def0996f3f5518f36c6919636d9209
# Author: Olli Pettay <Olli.Pettay@helsinki.fi>
# Date: 2020-02-07 21:53:54
# Regressor Bug: 1578628
# File Overlap Count: 2
# Description:
#   Bug 1612431, it is enough to update existing browsing context name when changing iframe.name, r=kmag
#   
#   This brings back some of the behavior we had before bug 1578628.
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D61637
# ==============================================================================

diff -r 60f14bea15af -r 656b57e594de dom/html/nsGenericHTMLFrameElement.cpp
--- a/dom/html/nsGenericHTMLFrameElement.cpp	Fri Feb 07 14:15:33 2020 +0100
+++ b/dom/html/nsGenericHTMLFrameElement.cpp	Thu Feb 06 18:44:05 2020 +0000
@@ -341,7 +341,7 @@
     } else if (aName == nsGkAtoms::name) {
       // Propagate "name" to the browsing context per HTML5.
       RefPtr<BrowsingContext> bc =
-          mFrameLoader ? mFrameLoader->GetBrowsingContext() : nullptr;
+          mFrameLoader ? mFrameLoader->GetExtantBrowsingContext() : nullptr;
       if (bc) {
         if (aValue) {
           bc->SetName(aValue->String());